Patch origin: in-house
Patch status: Solaris-specific; not suitable for upstream

Starts Tomcat in desired mode. See:
https://tomcat.apache.org/tomcat-6.0-doc/config/systemprops.html

--- bin/catalina.sh
+++ bin/catalina.sh
@@ -285,6 +285,7 @@
         -Dcatalina.base="$CATALINA_BASE" \
         -Dcatalina.home="$CATALINA_HOME" \
         -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+        -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
         org.apache.catalina.startup.Bootstrap "$@" start
     else
       exec "$_RUNJDB" "$LOGGING_CONFIG" $LOGGING_MANAGER $JAVA_OPTS $CATALINA_OPTS \
@@ -293,6 +294,7 @@
         -Dcatalina.base="$CATALINA_BASE" \
         -Dcatalina.home="$CATALINA_HOME" \
         -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+        -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
         org.apache.catalina.startup.Bootstrap "$@" start
     fi
   fi
@@ -312,6 +314,7 @@
       -Dcatalina.base="$CATALINA_BASE" \
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+      -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
       org.apache.catalina.startup.Bootstrap "$@" start
   else
     exec "$_RUNJAVA" "$LOGGING_CONFIG" $LOGGING_MANAGER $JAVA_OPTS $CATALINA_OPTS \
@@ -319,6 +322,7 @@
       -Dcatalina.base="$CATALINA_BASE" \
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+      -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
       org.apache.catalina.startup.Bootstrap "$@" start
   fi
 
@@ -376,6 +380,7 @@
       -Dcatalina.base="$CATALINA_BASE" \
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+      -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
       org.apache.catalina.startup.Bootstrap "$@" start \
       >> "$CATALINA_OUT" 2>&1 &
 
@@ -385,6 +390,7 @@
       -Dcatalina.base="$CATALINA_BASE" \
       -Dcatalina.home="$CATALINA_HOME" \
       -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+      -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
       org.apache.catalina.startup.Bootstrap "$@" start \
       >> "$CATALINA_OUT" 2>&1 &
 
@@ -435,6 +441,7 @@
     -Dcatalina.base="$CATALINA_BASE" \
     -Dcatalina.home="$CATALINA_HOME" \
     -Djava.io.tmpdir="$CATALINA_TMPDIR" \
+    -Dorg.apache.catalina.STRICT_SERVLET_COMPLIANCE=true \
     org.apache.catalina.startup.Bootstrap "$@" stop
 
   if [ ! -z "$CATALINA_PID" ]; then
